src/components/page/page_header/page_header_content.tsx

```diff
@@ -344,25 +344,34 @@ export const EuiPageHeaderContent: FunctionComponent<EuiPageHeaderContentProps>
 
   let rightSideFlexItem;
   if (rightSideItems && rightSideItems.length) {
-    const wrapWithFlex = () => {
-      return rightSideItems.map((item, index) => {
-        return (
-          <EuiFlexItem grow={false} key={index}>
-            {item}
-          </EuiFlexItem>
-        );
-      });
+    const wrapWithFlex = (reverse: boolean) => {
+      const reverseRightSideItems = reverse
+        ? rightSideItems.reverse()
```

Comment:
`.reverse()` mutates the array in place. Do not ever call it without shallow cloning the array first, especially on arrays coming from consumers (vs arrays we define/control).

```suggestion
      const reverseRightSideItems = reverse
        ? [...rightSideItems].reverse()
```

Reply:
> Do not ever call it without shallow cloning the array first

Thank you for this and duly noted!

Reply:
To clarify, I was being a little didactic with the initial "do not ever". If we're the ones defining an array and know we want to simply mutate/reverse the array in place, great, mutate away (although some schools of thought in JS hate this :).

In this case `rightSideItems` is not an array we define/own and we do not want ever want to mutate consumer data.